re-organized files

msleckman opened the pull request:
Adhering to the new folder structure and including 1_fetch. Moved the my_happy_script.R to `1_fetch/src/`

msleckman committed:
re-organized files

msleckman committed:
modified script plotting  model rmse data into functions

msleckman committed:
updated .gitignore to ignore script outputs

msleckman committed:
modified script plotting  model rmse data into functions

msleckman committed:
modified script plotting  model rmse data into functions

msleckman commented:
How to run the functions

```
downloaded_data_path <- data_extraction(filename_output_csv = csv_filename_output,
                                        download_data_location = in_path,
                                        selected_id_sciencebase = sb_id_sciencebase)

data <- prep_model_data(downloaded_data_path,
                        cleaned_data_location = out_path)

plot_model_data(data,
                output_plot_location = out_path)

render_data_diag(data)

```

msleckman committed:
modified script plotting  model rmse data into functions

msleckman committed:
modified script plotting  model rmse data into functions

msleckman committed:
modified script plotting  model rmse data into functions

jesse-ross commented on 1_fetch/src/my_happy_script_ms_11-12-21.R:
```diff
@@ -0,0 +1,196 @@
+## Global Variables:
+
+in_path = '1_fetch/src' 
+out_path = '1_fetch/out'
```
The script will not currently run if someone clones your repository and tries to run the code, because this directory isn't in git. One approach would be to write code to check whether the directory exists, and create it if it doesn't. But it is more elegant just to ensure that the directory shows up for others by committing a `.empty` file as detailed in my general comment.

msleckman replied:
hmm, i realized I wrote in my `.gitignore` `*/out/*` to exclude all files in out folder. Wouldn't that mean that even if I include a `.empty` in the `out` folder, the `out` will not appear in the repo? 
Instead, I can just replace `*/out/*` in my gitignore with calls to exclude the specific output files.

jesse-ross replied:
Yes, that works. But it does require you to potentially keep editing the `.gitignore` as you create outputs of new names. Another option is to keep the `*/out/*` in your `.gitignore` and then force git to commit the `.empty` file anyways by adding the `-f` flag like this: `git commit -f 1_fetch/out/.empty`

jesse-ross commented on 1_fetch/src/my_happy_script_ms_11-12-21.R:
```diff
@@ -0,0 +1,196 @@
+
+## Global Variables:
+
+in_path = '1_fetch/src' 
```
This variable is not used again, and can be removed.

jesse-ross commented on 1_fetch/src/my_happy_script_ms_11-12-21.R:
```diff
@@ -0,0 +1,196 @@
+
+in_path = '1_fetch/src' 
+out_path = '1_fetch/out'
+csv_filename_output = 'model_RMSEs.csv'
+sb_id_sciencebase = '5d925066e4b0c4f70d0d0599'
```
While R allows the use of `=` for assignment, many R programmers find it confusing. The convention in our group is to use `<-` instead. Can you please change the code throughout this file to use `<-` for assignment instead of `=`?

More generally, we officially use the [Tidyverse style guide](https://style.tidyverse.org/), and this is worth perusing. Our adherence to it is not perfect but it's a guiding light :)

msleckman replied:
Great, yes sticking with the `<-` for assignment then.

msleckman replied:
think i got all of them. let me know if i missed some.

jesse-ross commented on 1_fetch/src/my_happy_script_ms_11-12-21.R:
```diff
@@ -0,0 +1,196 @@
+eval_data = prep_model_data(downloaded_data_file_path = downloaded_data_path,
+                            cleaned_data_location = out_path)
+
+Plot_model_data(data = eval_data, output_plot_location = out_path)
```
The function is defined with a lower-case p as `plot_model_data` so this will not run!

msleckman replied:
Yes, I meant for this to be all lower case. Thanks for catching that!

jesse-ross commented on 1_fetch/src/my_happy_script_ms_11-12-21.R:
```diff
@@ -0,0 +1,196 @@
+### RUN ###
+
+downloaded_data_path <- data_extraction(filename_output_csv = csv_filename_output,
+                                        download_data_location = in_path,
```
Shouldn't this be `out_path` instead of `in_path`? The ScienceBase file you download is an output of the `1_fetch/` phase of the processj.

msleckman replied:
Since I saw this as a single script with 4 functions in the 1_fetch folder, it made more sense for me to put this in `in_path` since the downloaded data is in remains in its raw form and is then processed. However, split the script into two with the second saved in the `2_process` folder will allow me to put this into `1_fetch/out` and not make this so confusing. 👍

jesse-ross replied:
Oh, I see now, thanks! That makes perfect sense with the single-phase design.

jesse-ross reviewed (commented):
Hi Margaux, good work splitting this script into logical functions! There are a few more changes that I'd like you to make.

First, the only folder I'm seeing in your PR is the `1_fetch` folder. I'd like to also see the `2_process` folder (and a `3_visualize` folder if you choose to go that path), even if those folders are empty. Since git ignores empty directories, a good convention for doing this is to create empty files and then commit them, along the lines of 
```
touch 1_src/out/.empty
touch 2_process/out/.empty
```
Since the filenames I've given above start with a `.`, they won't show up in `ls`, but since they exist they will allow git to commit the directory structure.

Now that the directory structure will be showing up in the PR, can you also please split the script into multiple files, with the functions going in the phase that seems best? For instance your `data_extraction()` is simply fetching data, so it seems like it belongs in `1_fetch/src/` where it currently resides. But once that function has run, the other functions are all about processing and visualizing data, so they should go in different `*/src/` folders.

Once you have your changes ready, please tag me to re-review them - and don't hesitate to ask any questions or point out any areas where you disagree with my comments! You can do that through this PR process if you want, or through chat. Normally we use the PR process extensively for these kinds of discussions, but if a question seems worth posting to the group chat then please do so.. Thanks!

msleckman committed:
updates to the repo, split script into fetch and process script, added visualize folder

msleckman committed:
updates to the repo, split script into fetch and process script, added visualize folder

msleckman committed:
removed 3_visualize folder, and edited scripts

msleckman committed:
quick edit

msleckman commented:
Update - Running functions - example: 
```
downloaded_data_path <- data_extraction(filename_output_csv = csv_filename_output,
                                        download_data_location = out_path,
                                        selected_id_sciencebase = sb_id_sciencebase)

eval_data <- prep_model_data(output_data_location =  process_out_path)

plot_model_data(data = eval_data, output_plot_location = process_out_path)

render_data_diag(data = eval_data, output_path = process_out_path)
```

jesse-ross commented:
Hi Margaux - this does the job of splitting the phases, but there are a few changes I'd like to see before we move on. Can you please push them directly to the `main` branch and then let me know so I can review again? Push directly to `main` instead of creating another pull request - I'm afraid that a new pull request might confuse the bot.

Once I've had another look, then we can move on to the next issue. Thanks!

### `2_process/src/ModelPlotting_ms.R`
* line 24: `file.exists()` will return a boolean, so there is no need to test whether its result is equal to `TRUE`. It will be clearer just to express this along the lines of `if (file.exists(the_path_to_your_file) {do_some_stuff}`
* line 34: I like where you're going with this check for a missing input file, and auto-correction if it's missing. However there is a syntax error which is keeping the script from running!
* line 37: If this half of your if-else runs, then `downloaded_data_path` won't be set because it's only set in the first half, and this will probably cause problems further along.

msleckman commented:
> ### `2_process/src/ModelPlotting_ms.R`
> 
>     * line 24: `file.exists()` will return a boolean, so there is no need to test whether its result is equal to `TRUE`. It will be clearer just to express this along the lines of `if (file.exists(the_path_to_your_file) {do_some_stuff}`
Fixed by removing the `== True`

>     * line 34: I like where you're going with this check for a missing input file, and auto-correction if it's missing. However there is a syntax error which is keeping the script from running!
Fixed, removed the extra `)`
 
>     * line 37: If this half of your if-else runs, then `downloaded_data_path` won't be set because it's only set in the first half, and this will probably cause problems further along.
Fixed, `downloaded_data_path` no created yet at this stage in the script

msleckman commented:
**Solutions with Make:**
* stand alone software tool
* creates a makefile which is a config file that keeps record of the rules, linkages between project files, and dependencies in the project. It map out how the target file is created from source files
* Not specific to any programming language

** Solutions with R Targets:**
* A pipeline tool within R 
* Built for the R workflow
* `_targets.R` is the config file with rules and definition of the pipeline, equivalent to the makefile

jesse-ross commented:
Exactly. One of the best things about `targets` is the fact that it exists in R. That means that it can read the code and compare whether the code has _functionally_ changed. This means that if you go back and change formatting or add comments, `targets` will know that it doesn't need to re-run that code! It's a big advancement over the way that `make` just monitors for changes at the file level.
